### PR TITLE
Period granularity

### PR DESCRIPTION
**Pull request by Gauravshah**
added ability to specify a period granularity for time period to query on.

Reference:
https://druid.io/docs/latest/querying/granularities.html#period-granularities

screen shot 2017-08-03 at 3 18 24 am

### CODE REVIEW

File: dist/datasource.js [hunk 1]
```diff
@@ -102,6 +103,11 @@ function (angular, _, dateMath, moment) {
         //Round up to start of an interval
         //Width of bar chars in Grafana is determined by size of the smallest interval
         var roundedFrom = granularity === "all" ? from : roundUpStartTime(from, granularity);
+        if(dataSource.periodGranularity!=""){
+            if(granularity==='day'){
```
Comment (Gauravshah) [inline dist/datasource.js]: can be then later used for month as well

**Comment by abhisant** [comment]
Thanks @Gauravshah for contributing. Will review / test it and get back to you.

File: dist/datasource.js [hunk 2]
```diff
@@ -102,6 +103,11 @@ function (angular, _, dateMath, moment) {
         //Round up to start of an interval
         //Width of bar chars in Grafana is determined by size of the smallest interval
         var roundedFrom = granularity === "all" ? from : roundUpStartTime(from, granularity);
+        if(dataSource.periodGranularity!=""){
```
Comment (Gauravshah) [inline dist/datasource.js]: i guess adding a safety check for undefined would be better

**Comment by Gauravshah** [comment]
please do not merge it. It has an issue with the start time of the graph. Will work on it and get back in couple of days.

**Comment by abhisant** [comment]
Sure.

**Comment by Gauravshah** [comment]
can I switch rounding up like this:
```
    function roundUpStartTime(from, granularity) {
      var duration = _.find(GRANULARITIES, function (gEntry) {
        return gEntry[0] === granularity;
      })[1];
      var rounded = null;
      if(granularity=='day'){
          rounded = moment(+from).startOf('day');
      }else{
          rounded = moment(Math.ceil((+from)/(+duration)) * (+duration));
      }

      console.log("Rounding up start time from " + from.format() + " to " + rounded.format() + " for granularity [" + granularity + "]");
      return rounded;
    }

```

**Comment by Gauravshah** [comment]
@abhisant do you know of a way to check the timezone setting via plugin ? I guess that would be the right thing to do. Look at the time zone setting and then set start of day either to local time or UTC based on what is in grafana settings

**Comment by Gauravshah** [comment]
https://github.com/grafana/grafana/pull/8916 this will fix it. Will test once that is merged and reply back

**Comment by abhisant** [comment]
Sorry, @Gauravshah : Didn't get a chance to respond back to your queries. Thanks for the update.

**Comment by Gauravshah** [comment]
grafana's pull request is merged, and druid plugin needs to resolve it in a different way. 
@abhisant can you please review & merge this one ? 
Will create another pull request in few weeks for fixing the start time of graph

**Pull request merged**